src/platforms/linux-raspbian.js

```diff
@@ -131,6 +131,293 @@ function setLanMode(mode, options = {}) {
+  );
+  if (proc.status === 0) {
+    const networks = proc.stdout.split('\n')
+      .filter((l) => !l.startsWith('network'))
+      .map((l) => l.split(' ')[0])
+      .reverse();
+
+    for (const id of networks) {
+      proc = child_process.spawnSync(
+        'wpa_cli',
+        ['-i', 'wlan0', 'remove_network', id]
+      );
+      if (proc.status !== 0) {
+        console.log('Failed to remove network with id:', id);
+      }
+    }
+  }
+
+  // Then, disable hostapd. It will either need to be off or reconfigured, so
+  // this is valid in both modes.
+  proc = child_process.spawnSync(
+    'sudo',
+    ['systemctl', 'stop', 'hostapd.service']
+  );
+  if (proc.status !== 0) {
+    return false;
+  }
+
+  proc = child_process.spawnSync(
+    'sudo',
+    ['systemctl', 'disable', 'hostapd.service']
+  );
+  if (proc.status !== 0) {
+    return false;
+  }
+
+  if (!enabled) {
+    return true;
+  }
+
+  // Now, set the IP address back to a sane state
+  proc = child_process.spawnSync(
+    'sudo',
+    ['ifconfig', 'wlan0', '0.0.0.0']
+  );
+  if (proc.status !== 0) {
+    return false;
+  }
+
+  if (mode === 'sta') {
+    proc = child_process.spawnSync(
+      'wpa_cli',
+      ['-i', 'wlan0', 'add_network'],
+      {encoding: 'utf8'}
+    );
+    if (proc.status !== 0) {
+      return false;
+    }
+
+    const id = proc.stdout.trim();
+
+    options.ssid = options.ssid.replace('"', '\\"');
+    proc = child_process.spawnSync(
+      'wpa_cli',
+      // the ssid argument MUST be quoted
+      ['-i', 'wlan0', 'set_network', id, 'ssid', `"${options.ssid}"`]
+    );
+    if (proc.status !== 0) {
+      return false;
+    }
+
+    if (options.key) {
+      options.key = options.key.replace('"', '\\"');
+      proc = child_process.spawnSync(
+        'wpa_cli',
+        // the psk argument MUST be quoted
+        ['-i', 'wlan0', 'set_network', id, 'psk', `"${options.key}"`]
+      );
+    } else {
+      proc = child_process.spawnSync(
+        'wpa_cli',
+        ['-i', 'wlan0', 'set_network', id, 'key_mgmt', 'NONE']
+      );
+    }
+
+    if (proc.status !== 0) {
+      return false;
+    }
+
+    proc = child_process.spawnSync(
+      'wpa_cli',
+      ['-i', 'wlan0', 'enable_network', id]
+    );
+    if (proc.status !== 0) {
+      return false;
+    }
+
+    proc = child_process.spawnSync(
+      'wpa_cli',
+      ['-i', 'wlan0', 'save_config']
+    );
+    if (proc.status !== 0) {
+      return false;
+    }
+  } else {
+    let config = 'interface=wlan0\n';
+    config += 'driver=nl80211\n';
+    config += 'hw_mode=g\n';
+    config += 'channel=6\n';
+    config += `ssid=${options.ssid}\n`;
+
+    if (options.key) {
+      config += 'wpa=2\n';
+      config += `wpa_passphrase=${options.key}\n`;
+      config += 'wpa_pairwise=CCMP\n';
+    }
+
+    fs.writeFileSync('/tmp/hostapd.conf', config);
+
+    proc = child_process.spawnSync(
+      'sudo',
+      ['mv', '/tmp/hostapd.conf', '/etc/hostapd/hostapd.conf']
+    );
```

Comment:
/tmp and /etc are almost certainly on different filesystems. Which means that the mv will actually be a copy, which isn't atomic. I think it would be better to make the temporary file be /etc/hostspd.conf.new and then do the mv

Then the move will be atomic and you'll get either the entire new file or the entire old file, and not half of the new file if a crash occurs at the wrong time.

Reply:
And then you could use fs.renameSync rather than having to launch a process.

Reply:
The problem is that the gateway runs as non-root on Raspbian, but writing to `/etc` requires root access.

Reply:
Therefore, we can't directly use the `fs` methods to write to `/etc`.

Reply:
ok - then perhaps do a sudo cp to copy it into /etc and then do a sudo mv to rename it.

Reply:
These could be combined using something like sudo bash -c 'cp blah && mv blah'

Reply:
In our Raspbian image, `/etc` and `/tmp` are both on the same device, `/dev/mmcblk0p2`.

Reply:
ok - I'm convinced. I hadn't checked specifically, and I know many small distros use a ram disk for /tmp